### ISSUE TITLE
[Bug]: Sync download blobs already available in the local repository

### ISSUE THREAD

**Issue by AlbanBedel** [project-zot/zot#2661]
### zot version

v2.1.1

### Describe the bug

The sync code always download every blobs behind a tag even if they are already available locally.

### To reproduce

1. Setup zot as pass through cache
2. Inspect an image index, for example using `skopeo inspect docker://registry.example.com/repo/image:v1.2.3`
3. Inspect one of image referenced in the index, for example using `skopeo inspect docker://registry.example.com/repo/image:v1.2.3-arm64`
4. Notice how slow it is although all blobs are already in storage, or check the `repo/image/.sync/` directory to see that it download existing blobs again.

### Expected behavior

Blobs from the same repository are not downloaded again when already available in local storage.

**Comment by rchincha** [comment]
@AlbanBedel We take into account the case where images are being tagged with the same tag, meaning image with digest1 tagged with "tag1", oops!, pushed image with digest2 but retagged as same "tag1".

That said, we do make sure blobs/manifests don't get copied over if they already exist locally.

Is this not what you are seeing?

**Comment by AlbanBedel** [comment]
The case here is to pull an index tag that references blobs A,B,C and D. Then pull an image tag that reference only A and B. In this case A and B get downloaded again, with larger blobs this is very obvious alone from the time it take.

**Comment by andaaron** [comment]
Hi @AlbanBedel, are those docker images or OCI images at docker://registry.example.com/repo/image? Or can you send us a link to the upstream image for which this issue reproduces?

For docker format we convert the manifest and config blobs locally. It does not however explain not reusing the blobs which contain the image file system.

**Comment by eusebiu-constantin-petu-dbk** [comment]
Yes you are right, we just compare the parent digest (index digest in this case), we don't check each child...

**Comment by andaaron** [comment]
@eusebiu-constantin-petu-dbk, is this fixed with the change in https://github.com/project-zot/zot/pull/2903?

**Comment by andaaron** [comment]
Closing this as there hasn't been activity for a while, and sync was completely rewritten in the meantime.
Please reopen if needed.
